src/sources/envoy_als/grpc.rs

```diff
@@ -33,68 +37,99 @@ impl AccessLogService for Service {
             tokio::select! {
                 _ = &mut shutdown => break,
                 stream_msg = in_stream.message() => {
-                    match stream_msg {
-                        Ok(omsg) => {
-                            match omsg {
-                                Some(msg) => {
-                                    let mut events = vec![];
+                    match self.handle_message(stream_msg, &mut stream_identifier).await {
+                        Ok(_) => {},
+                        Err(err) => {
+                            match err {
+                                StreamError::StreamClosed => break,
```

Comment:
Actually, am I taking it to understand that the stream closing is not an error condition?
Because by breaking here, we are returning from this function with `Ok`

Reply:
Correct. This is in the case of a graceful stream close by Envoy.